tot/data_processing/scaler.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+
+import numpy as np
+import pandas as pd
+
+
+@dataclass
+class Scaler:
+    transformer: object
+
+    def __post_init__(self):
+        is_transformer_valid = (
+            callable(getattr(self.transformer, "fit", None))
+            and callable(getattr(self.transformer, "transform", None))
+            and callable(getattr(self.transformer, "inverse_transform", None))
+        )
+        if not is_transformer_valid:
+            raise ValueError(
+                "Transformer provided to the Scaler must implement fit, transform and inverse_transform " "methods"
+            )
+
+    def _transform(self, df: pd.DataFrame, fit):
+        values = []
+        for df_name, df_i in df.groupby("ID"):
```

Comment:
For loops have nice readability. However, they consume computing time. Can you rewrite that part using `.apply()`?

Reply:
I think we might be able to directly pass a df instead of converting it to a list. :) Have a look at this post: https://stackoverflow.com/questions/35723472/how-to-use-sklearn-fit-transform-with-pandas-and-return-dataframe-instead-of-num

Reply:
I changed the implementation to use `pd.pivot()` bc I'm not sure how to use `.apply()` in this context